src/diffusers/loaders/transformer_sd3.py

```diff
@@ -0,0 +1,261 @@
+import inspect
+from functools import partial
+from typing import Dict, List, Optional, Union
+
+import torch.nn as nn
+
+from ..utils import (
+    USE_PEFT_BACKEND,
+    delete_adapter_layers,
+    is_accelerate_available,
+    logging,
+    set_adapter_layers,
+    set_weights_and_activate_adapters,
+)
+from .lora import TEXT_ENCODER_NAME, TRANSFORMER_NAME
+
+
+if is_accelerate_available():
+    from accelerate.hooks import AlignDevicesHook, CpuOffload, remove_hook_from_module
+
+logger = logging.get_logger(__name__)
+
+
+class SD3TransformerLoadersMixin:
```

Comment:
Is this basically on the same level as `LoraLoaderMixin` but it doesn't make sense to inherit from `LoraBaseMixin`?

Reply:
No, so this is analogous to https://github.com/huggingface/diffusers/blob/main/src/diffusers/loaders/unet.py, which implements the LoRA-level functionalities at the model level. 

`LoraBaseMixin` is for pipelines. 

Hence I stated in the PR description:

> Currently, we re-implement some methods in both UNet2DLoaderMixin and SD3TransformerLoaderMixin with the "# Copied from ..." mechanism. In a future PR, we can consider introducing a ModelLoaderMixin so that we can share methods like set_adapter() on the ModelMixin level. This should be relatively easy to do.